iceoryx_integrationtest/src/iox_publisher_integrationtest.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/typed_publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/cxx/smart_c.hpp"
+
+#include <iostream>
+
+std::atomic_bool killSwitch{false};
+
+static void sigHandler(int32_t signal [[gnu::unused]])
+{
+    killSwitch.store(true);
+}
+
+void registerSigHandler()
+{
+    // register sigHandler for SIGINT, SIGTERM and SIGHUP
+    struct sigaction act;
+    sigemptyset(&act.sa_mask);
+    act.sa_handler = sigHandler;
+    act.sa_flags = 0;
+
+    if (iox::cxx::makeSmartC(sigaction, iox::cxx::ReturnMode::PRE_DEFINED_SUCCESS_CODE, {0}, {}, SIGINT, &act, nullptr)
```

Comment:
Do you know if the sigaction struct must live until the end of the program? I'm pretty sure it must not, but I found only examples where the struct is instantiated in `main` and therefore also lives until the end of the program.

Reply:
There is nothing mentioned in: https://man7.org/linux/man-pages/man2/sigaction.2.html But maybe I have overlooked it.

I suspect it is only a pointer since by reference arguments are not allowed in C and this is maybe somehow larger?!